frontend/src/GameReportForm.tsx

```diff
@@ -625,4 +911,28 @@ function VictoryPoints({ strongholds }: VictoryPointsProps) {
     );
 }
 
+function detectMissingInput(value: unknown): FieldError {
+    return value !== null &&
+        value !== undefined &&
+        value !== "" &&
+        (typeof value === "number" ? value >= 0 : true)
```

Comment:
Not in this PR, but maybe `number` fields should be changed to `number | null` instead of having `-1` represent an empty input? That would be parallel to the `string | null` fields. Alternatively, could make the `string | null` fields only `string` and have `""` represent an empty input for strings. Any preference?

Reply:
Yes, 100%. I'm not sure why I did it this way, probably because I was trying to keep some fields non-nullable. But I was conflating UI representation (e.g. `gameTurns` could be `null`) with data representation (`gameTurns` is a non-null int >= 1). That can be fixed now.

I probably prefer null for no input (absence of a value)? Just a slight preference because it mirrors the other fields that are missing their values. No strong opinion though.

Reply:
Been there haha. With you on `null`, let's do it!